hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+    '''
+    abscal_delta_gains = {ant: np.ones_like(g, dtype=complex) for ant, g in rc_flags.items()}
+
+    # instantiate Abscal object
+    wgts = DataContainer({k: (~flags[k]).astype(np.float) for k in flags.keys()})
+    AC = AbsCal(model, data, wgts=wgts, antpos=data.antpos, freqs=data.freqs, 
+                refant=pick_reference_antenna(synthesize_ant_flags(flags))[0],
+                min_bl_cut=min_bl_cut, max_bl_cut=max_bl_cut)
+    AC.antpos = data.antpos
+
+    # Global Delay Slope Calibration
+    for time_avg in [True, False]:
+        abscal_step(abscal_delta_gains, AC, AC.delay_slope_lincal, {'time_avg': time_avg, 'edge_cut': edge_cut, 'verbose': verbose},
+                    [AC.custom_dly_slope_gain], [(rc_flags.keys(), data.antpos)], rc_flags,
+                    gain_convention=gain_convention, verbose=verbose)
+
+    # Global Phase Slope Calibration
+    abscal_step(abscal_delta_gains, AC, AC.global_phase_slope_logcal, {'tol': tol, 'edge_cut': edge_cut, 'verbose': verbose},
+                [AC.custom_phs_slope_gain], [(rc_flags.keys(), data.antpos)], rc_flags,
+                gain_convention=gain_convention, max_iter=phs_max_iter, phs_conv_crit=phs_conv_crit, verbose=verbose)
+
+    # Per-Channel Absolute Amplitude Calibration
+    abscal_step(abscal_delta_gains, AC, AC.abs_amp_logcal, {'verbose': verbose}, [AC.custom_abs_eta_gain], 
+                [(rc_flags.keys(),)], rc_flags, gain_convention=gain_convention, verbose=verbose)
+
+    # Per-Channel Tip-Tilt Phase Calibration
+    abscal_step(abscal_delta_gains, AC, AC.TT_phs_logcal, {'verbose': verbose}, [AC.custom_TT_Phi_gain, AC.custom_abs_psi_gain], 
+                [(rc_flags.keys(), data.antpos), (rc_flags.keys(),)], rc_flags,
+                gain_convention=gain_convention, max_iter=phs_max_iter, phs_conv_crit=phs_conv_crit, verbose=verbose)
+
+    return abscal_delta_gains, AC
+
+
+def post_redcal_abscal_run(data_file, redcal_file, model_files, output_replace=('.omni.', '.abs.'), nInt_to_load=-1,
+                           data_solar_horizon=90, model_solar_horizon=90, min_bl_cut=1.0, max_bl_cut=None, edge_cut=0, 
+                           tol=1.0, phs_max_iter=100, phs_conv_crit=1e-6, clobber=True, add_to_history='', verbose=True):
+    '''Perform abscal on entire data files, picking relevant model_files from a list and doing partial data loading.
+    
+    Arguments:
+        data_file: string path to raw uvh5 visibility file
+        redcal_file: string path to calfits file that redundantly calibrates the data_file
+        model_files: list of string paths to externally calibrated data. Strings must be sortable 
+            to produce a chronological list in LST (wrapping over 2*pi is OK)
+        output_replace: tuple of two strings to find and replace in redcal_file to produce the output calfits file
+        nInt_to_load: number of integrations to load and calibrate simultaneously. Default -1 loads all integrations.
```

Comment:
can this be `None` instead of -1, I'm used to `-1` being an axis specification, and also `None` follows `io.HERAData` convention when specifying `bls` `pols` etc..

Reply:
Done. I also updated redcal to have a consistent convention.